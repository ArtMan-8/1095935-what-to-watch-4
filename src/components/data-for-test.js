export const moviePoster = {
  title: `Seven Years in Tibet`,
  genre: `Drame`,
  year: `1997`,
};

export const films = [{
  id: `1184`,
  title: `We need to talk about Kevin`,
  poster: `img/we-need-to-talk-about-kevin.jpg`,
},
{
  id: `1134`,
  title: `What We Do in the Shadows`,
  poster: `img/what-we-do-in-the-shadows.jpg`,
},
{
  id: `1148`,
  title: `Revenant`,
  poster: `img/revenant.jpg`,
},
{
  id: `1194`,
  title: `Johnny English`,
  poster: `img/johnny-english.jpg`,
}];
