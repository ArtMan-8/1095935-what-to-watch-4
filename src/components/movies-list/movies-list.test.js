import React from 'react';
import renderer from 'react-test-renderer';
import MoviesList from './movies-list.jsx';
import {films} from '../data-for-test.js';

describe(`MoviesList`, () => {
  it(`Render MoviesList`, () => {
    const tree = renderer.create(
        <MoviesList
          films={films}
          onSmallMovieCardClick={() => {}}
        />, {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
