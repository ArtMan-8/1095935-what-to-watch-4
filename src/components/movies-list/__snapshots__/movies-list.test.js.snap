// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList Render MoviesList 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="We need to talk about Kevin"
        height="175"
        src="img/we-need-to-talk-about-kevin.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page"
        onClick={[Function]}
      >
        We need to talk about Kevin
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="What We Do in the Shadows"
        height="175"
        src="img/what-we-do-in-the-shadows.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page"
        onClick={[Function]}
      >
        What We Do in the Shadows
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Revenant"
        height="175"
        src="img/revenant.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page"
        onClick={[Function]}
      >
        Revenant
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Johnny English"
        height="175"
        src="img/johnny-english.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page"
        onClick={[Function]}
      >
        Johnny English
      </a>
    </h3>
  </article>
</div>
`;
